Integrate serial communication functionality

mvoodarla opened the pull request:
This adds serial functionality to the robot code, works consistently after testing on 1/17. CV angle and distance is accessible through the ``Devices.java`` class.

mvoodarla committed:
Add all serial communication code

mvoodarla committed:
Test working serial communication

mvoodarla committed:
Centralize all serial communication into devices class

Vision values are updated in the periodic methods in Robot.java but
everything stays inside Devices.java

mvoodarla committed:
Make changes to create working serial connection in Devices class

AG6GR reviewed (changes requested):
Overall look alright, but serial is managing enough state that I think it is best in its own module/class. Especially if we decide to do more fancy behavior later.

mvoodarla committed:
Move serial communication to its own singleton class

JetsonPort class will have all functionality required to interface with
Jetson over socket connection.

mvoodarla committed:
Make JetsonPort constructor private
